src/test/groovy/life/qbic/utils/BioinformaticAnalysisSpec.groovy

```diff
@@ -0,0 +1,62 @@
+package life.qbic.utils
+
+import spock.lang.Specification
+import java.nio.file.NotDirectoryException
+import java.nio.file.Paths
+import java.text.ParseException
+
+/**
+ *  Tests for the BioinformaticAnalysisParser
+ *
+ * @since 1.8.0
+ * @see BioinformaticAnalysisParser
+ *
+ */
+class BioInformaticAnalysisSpec extends Specification {
+
+    def exampleDirectoriesRoot = this.getClass().getResource("/dummyFileSystem/bioinformatic-analysis-output").getPath()
+    BioinformaticAnalysisParser bioinformaticAnalysisParser = new BioinformaticAnalysisParser()
+
+    def "parsing a valid file structure returns a Json String"() {
+        given:
+        def pathToDirectory = Paths.get(exampleDirectoriesRoot, "validates")
+        when:
+        bioinformaticAnalysisParser.parseFrom(pathToDirectory)
+        then:
+        noExceptionThrown()
+    }
+
+    def "parsing an empty directory throws ParseException"() {
+        given:
+        def pathToDirectory = Paths.get(exampleDirectoriesRoot, "empty_directory/")
+        // Maven doesn't include empty folders in build process so it has to be generated explicitly
+        File directory = new File(pathToDirectory.toString())
+        if (!directory.exists()) {
+            directory.mkdir()
+        }
+        when:
+        bioinformaticAnalysisParser.parseFrom(pathToDirectory)
+        then:
+        thrown(ParseException)
+        // Remove new created folder after testing
+        directory.delete()
+    }
```

Comment:
Since we don't have more granular exception classes, i suggest to check the message for the one that is expected here.

Reply:
Good point, will do 👍